### ISSUE TITLE
Boolean operators return the wrong type

### ISSUE THREAD

**Issue by ccapndave** [custom-components/pyscript#335]
Whilst trying to make a battery sensor, I stumbled on some very strange behaviour (I'm doing this in a Jupyter notebook using https://github.com/craigbarratt/hass-pyscript-jupyter).

The script at the top should return an empty list, but instead it returns a list.  I explored further and there seems to be something strange about the boolean `and` operator.  Here is what's in my notebook:

```python
import re

state_names = ['sun.sun', 'group.battery_level', 'group.battery_alert', 'input_number.battery_alert_threshold_max']

ignore_sensors = ["sun.sun", "group.battery_level"]
[sensor for sensor in state_names if (re.search(r'_battery(_level)?$', sensor) and (sensor not in ignore_sensors))]
```

    ['group.battery_alert', 'input_number.battery_alert_threshold_max']
---
```python
[re.search('battery', sensor) and sensor not in ignore_sensors for sensor in state_names]
```

    [0, 0, True, True]

---
```python
type(True and False)
```

    <class 'int'>

---
```python
type(False)
```

    <class 'bool'>

---
```python
type(True)
```

    <class 'bool'>

---
```python
type(True and True)
```

    <class 'bool'>

---
```python
type(True and True)
```

    <class 'bool'>

---
```python
bool(0)
```

    False

**Comment by vaizki** [comment]
Checked out https://github.com/custom-components/pyscript/blob/master/custom_components/pyscript/eval.py#L1678 and it seems the following "logic" applies for `A and B`:
- if `A` or `B` is false-ish an `int` with value `0` is returned
- if `A` and `B` are true-ish, then `B` is returned
Would seem sensible to return a bool `False` or `True` instead of `0` and something that happened to be on the right side of `and`.

Same kind of logic happens with `or` in the same function.

**Comment by craigbarratt** [comment]
Thanks for find and reporting the bug.  I just pushed a fix.  If you get a chance, please test it to make sure it resolves your test cases.  I'll add some test cases too.

**Comment by craigbarratt** [comment]
This is fixed in 1.4.0.  Closing.
